pftools/python/parflow/tools/builders.py

```diff
@@ -534,5 +666,314 @@ def get_table(self, props_in_header=True, column_separator='  '):
         print(self.get_table(props_in_header, column_separator))
         return self
+
+# -----------------------------------------------------------------------------
+# Domain input builder - setting keys for various common problem definitions
+# -----------------------------------------------------------------------------
+
+
+def get_file_extension(file_name):
+    return file_name.split('.')[-1]
+
+
+class DomainBuilder:
+
+    def __init__(self, run, name='domain'):
+        self.run = run
+        self.run.Domain.GeomName = name
+
+    def water(self, geom_name=None):
+        """Setting keys for water properties and gravity
+        """
+        self.run.Gravity = 1.0
+        self.run.Phase.Names = 'water'
+        self.run.Phase.water.Density.Type = 'Constant'
+        self.run.Phase.water.Density.Value = 1.0
+        self.run.Phase.water.Viscosity.Type = 'Constant'
+        self.run.Phase.water.Viscosity.Value = 1.0
+        self.run.Phase.water.Mobility.Type = 'Constant'
+        self.run.Phase.water.Mobility.Value = 1.0
+        self.run.PhaseSources.water.Type = 'Constant'
+
+        if geom_name:
+            self.run.PhaseSources.water.GeomNames = geom_name
+            self.run.PhaseSources.water.Geom[geom_name].Value = 0.0
+
+        return self
+
+    def no_wells(self):
+        """Setting key with no wells
+        """
+        self.run.Wells.Names = ''
+
+        return self
+
+    def no_contaminants(self):
+        """Setting key with no contaminants
+        """
+        self.run.Contaminants.Names = ''
+
+        return self
+
+    def variably_saturated(self):
+        """Setting keys for variably saturated domain.
+        Solver settings taken from default_richards test case
+        """
+        self.run.Solver = 'Richards'
+        self.run.Solver.MaxIter = 5
+        self.run.Solver.Nonlinear.MaxIter = 10
+        self.run.Solver.Nonlinear.ResidualTol = 1e-9
+        self.run.Solver.Nonlinear.EtaChoice = 'EtaConstant'
+        self.run.Solver.Nonlinear.EtaValue = 1e-5
+        self.run.Solver.Nonlinear.UseJacobian = True
+        self.run.Solver.Nonlinear.DerivativeEpsilon = 1e-2
+        self.run.Solver.Linear.KrylovDimension = 10
+        self.run.Solver.Linear.Preconditioner = 'PFMG'
+
+        return self
+
+    def fully_saturated(self):
+        """Fully saturated solver settings (other than solver ='Impes')
+        taken from default_richards test case - need to change
+        """
+        self.run.Solver = 'Impes'
+        self.run.Solver.MaxIter = 5
+        self.run.Solver.Nonlinear.MaxIter = 10
+        self.run.Solver.Nonlinear.ResidualTol = 1e-9
+        self.run.Solver.Nonlinear.EtaChoice = 'EtaConstant'
+        self.run.Solver.Nonlinear.EtaValue = 1e-5
+        self.run.Solver.Nonlinear.UseJacobian = True
+        self.run.Solver.Nonlinear.DerivativeEpsilon = 1e-2
+        self.run.Solver.Linear.KrylovDimension = 10
+        self.run.Solver.Linear.Preconditioner = 'PFMG'
+
+        return self
```

Comment:
@reedmaxwell @lecondon Do you have any suggestions of default solver settings here?

Reply:
@maxwell @lecondon Can you look at these default solver settings to see if they look reasonable?

Do people in practice solve to tolerances like this?   We use that in the 1e-9 in the regression tests but I'm wondering if we need that precision for real problems by default.